barretenberg/cpp/src/barretenberg/eccvm/eccvm_verifier.cpp

```diff
@@ -270,7 +270,7 @@ template <typename Flavor> bool ECCVMVerifier_<Flavor>::verify_proof(const plonk
         univariate_opening_verified = PCS::verify(pcs_verification_key, batched_univariate_claim, transcript);
     }
 
-    return multivariate_opening_verified && univariate_opening_verified;
+    return sumcheck_verified.value() && multivariate_opening_verified && univariate_opening_verified;
```

Comment:
Haven't we already verified the sumcheck by this time?

Reply:
Yeah but it felt safer to add this check (why not? also what if somehow the option doesn't have a value and the early return is  missed? not sure what happens then).